Fixed extension test skipping

timoti111 opened the pull request:
- when section had global extension need in tests.json and test was in file and was not defined in tests.json test was not skipped
- asc() test can now be run without FUNEXP extension
- chr() test can now be run without FUNEXP extension
- tests 09/04 and 09/14 need IFTHEN extension

timoti111 committed:
Fixed extension test skipping
- when section had global extension need in tests.json and test was in file and was not defined in tests.json test was not skipped

martinkobelka commented on ifj2017/tests/02_functions/tests.json:
```diff
@@ -14,13 +14,13 @@
       "info": "Asc value",
-      "stdout": " 104",
-      "code": "scope \n print asc(!\"ahoj\", 2); \n end scope"
+      "stdout": " 111",
```
Není pravda, zadej si na [https://www.jdoodle.com/execute-freebasic-online](url) 

/' Zajisteni zakladni kompatibility IFJ17->FreeBASIC '/

Function Length(s As String) As Integer
  Return Len(s)
End Function

Function SubStr(s as String, i As Integer, n As Integer) As String
  Return Mid(s, i, n)
End Function

/' Zde bude nasledovat program jazyka IFJ17 '/

scope
 print asc (!"ahoj", 2);
end scope

Výstup bude 104

timoti111 replied:
Bral som to zo zadania tak že druhý parameter znamená index v poli charov ale je pravda že v tomto prípade to dáva oveľa väčší zmysel keď je to brané ako pozícia v stringu. Opravené.

martinkobelka commented on ifj2017/tests/02_functions/tests.json:
```diff
@@ -14,13 +14,13 @@
-      "stdout": " 104",
-      "code": "scope \n print asc(!\"ahoj\", 2); \n end scope"
+      "stdout": " 111",
+      "code": "scope \n dim a as integer = asc(!\"ahoj\", 2) \n print a; \n end scope"
```
Není v základním zadání volání funkce povoleno pouze jako id = funkce(seznam_parametrů) ?

Ano moje zadání neodpovídalo ale řekl bych že tvé také ne. Mohu se mýlit.

thejoeejoee replied:
Souhlas s @martinkobelka, musí tam být samostatné přiřazení.

timoti111 replied:
Áno to je pravda. Opravené.

martinkobelka commented on ifj2017/tests/02_functions/tests.json:
```diff
@@ -14,13 +14,13 @@
       "info": "Chr function",
       "stdout": "a",
-      "code": "scope \n print chr(97); \n end scope"
+      "code": "scope \n dim s as string = chr(97) \n print s; \n end scope"
```
Stejný problém jako výše?

timoti111 replied:
Opravené.

thejoeejoee reviewed (changes requested):
Zatím nemohu vzít, některá očekávání v testech jsou chybná, viz komentáře.

timoti111 committed:
Fixed some tests
- removed need of FUNEXP extension on asc() and chr() tests
- tests 05/04 and 05/14 needs IFTHEN extension

thejoeejoee commented:
Díky @timoti111.
